src/pg/mod.rs

```diff
@@ -477,6 +479,7 @@ impl AsyncPgConnection {
             conn,
             None,
             None,
+            None,
```

Comment:
This is the function we could think about deprecating right? It does look weird having all these options. Can you share the rationale behind this construction addition to the project?

Reply:
I need the ability to use both `tokio_postgres::connect` and notifications, so there should be a constructor that takes both `Client` and `Connection`.

Reply:
@dullbananas That [method](https://docs.rs/diesel-async/latest/diesel_async/struct.AsyncPgConnection.html#method.try_from_client_and_connection) already exists, so I'm not sure what you want to add here.